src/main/java/org/truffleruby/core/numeric/FloatNodes.java

```diff
@@ -954,8 +955,9 @@ protected static boolean hasSmallExp(double value) {
             return (abs < 0.0001) && (abs != 0.0);
         }
 
-        protected static Rope specialValueRope(double value) {
-            return RopeOperations.encodeAscii(Double.toString(value), Encodings.US_ASCII.jcoding);
+        protected static TruffleString specialValueString(double value,
```

Comment:
Because this is only called by `@Cached` there is no advantage to pass a FromJavaStringNode, and it's better to use `TStringUtils#fromJavaString()` (less footprint).